lib/collection/src/collection_manager/optimizers/segment_optimizer.rs

```diff
@@ -543,7 +543,9 @@ pub trait SegmentOptimizer {
 
         if !all_segments_ok {
             // Cancel the optimization
-            return Ok(false);
+            return Err(CollectionError::Cancelled {
+                description: "Some segments are not ready for optimization".to_string(),
+            });
```

Comment:
I don't think we want to remove the boolean here. We need to propagate it through the callback in the match statement, and I don't think your alternative gives us exactly the same result.